ethcore/src/types/trace_types/filter.rs

```diff
@@ -108,27 +108,27 @@ impl Filter {
 
 	/// Returns true if given trace matches the filter.
 	pub fn matches(&self, trace: &FlatTrace) -> bool {
-		let action = match trace.action {
+		match trace.action {
 			Action::Call(ref call) => {
 				let from_matches = self.from_address.matches(&call.from);
 				let to_matches = self.to_address.matches(&call.to);
 				from_matches && to_matches
 			}
 			Action::Create(ref create) => {
 				let from_matches = self.from_address.matches(&create.from);
-				let to_matches = self.to_address.matches_all();
+
+				let to_matches = match trace.result {
```

Comment:
Hmm, now that I saw it's coming from `result` I'm not sure if it's equivalent.
@debris is it possible that `trace.action` is `Action::Call` but the result is `Res::Create` somehow?